0000-settlement-engines/0000-settlement-engines.md

```diff
@@ -0,0 +1,491 @@
+
+A **standard unit** is the typical unit of value for a particular asset, such as \$1 in the case of U.S. dollars, or 1 BTC in the case of Bitcoin.
+
+A **fractional unit** represents some unit smaller than the standard unit, but with greater precision. Examples of fractional monetary units include one cent (\$0.01 USD), or 1 satoshi (0.00000001 BTC).
+
+An **asset scale** is the difference in orders of magnitude between a standard unit and a corresponding fractional unit. More formally, the asset scale is a non-negative integer (0, 1, 2, …) such that one standard unit equals `10^(-scale)` of a corresponding fractional unit. If the fractional unit equals the standard unit, then the asset scale is 0.
+
+For example, one cent represents an asset scale of 2 in the case of USD, whereas 1 satoshi represents an asset scale of 8 in the case of Bitcoin.
+
+#### Selecting scales
+
+Settlement engines are RECOMMENDED to perform or fulfill requests with amounts denominated in the same scale as its settlements.
+
+Account balances within the accounting systems are RECOMMENDED to be denominated in the same scale as their corresponding settlement engine, but MAY use different ones. For example, micropayments may require more precision than can actually be settled, or databases may limit precision to less than the settlement system is capable of.
+
+#### `Quantity` object
+
+An amount denominated in some unit of a single, fungible asset. (Since each account is denominated in a single asset, the type of asset is implied.)
+
+##### Attributes
+
+- **`amount`** &mdash; string
+  - Quantity of the unit, which is a non-negative integer.
+  - This amount is encoded as a string to ensure no precision is lost on platforms that don't natively support arbitrary precision integers.
+- **`scale`** &mdash; number
+  - Asset scale of the unit, between `0` and the maximum 8-bit unsigned integer, `255` (inclusive).
+
+##### Example
+
+To represent $2.54 in units of cents, where the amount is a multiple of $0.01:
+
+```json
+{
+  "amount": "254",
+  "scale": 2
+}
+```
+
+#### Scale conversions
+
+If the accounting system or settlement engine receives a request with a **[`Quantity`](#quantity-json-type)** denominated in a unit more precise than its unit, it MUST convert the quantity into its native unit. If so, the resulting amount MUST be rounded down before fulfilling the request.
+
+The response to the request MUST include the converted, rounded **[`Quantity`](#quantity-json-type)** used to fulfill the request, which MUST be less than or equal to the amount sent in the original request. (If the amount rounds down to 0, this this amount MAY be 0.)
+
+Then, the system with the additional precision initiating the request MUST track the leftover sum so it may accumulate and be retried in subsequent requests.
+
+### Settlement Engine HTTP API Endpoints
+
+Settlement engines MUST implement these endpoints.
+
+#### Initiate an account
```

Comment:
Good idea, changed to "Setup an account." I think this is much more fitting since the account isn't "created" on the settlement engine -- it's shared between the two systems, this is just telling the SE that it was created.